jupyterhub/services/service.py

```diff
@@ -123,6 +123,8 @@ def user_env(self, env):
 
     def start(self):
         """Start the process"""
+        if self.port == 0:
+            self.port = random_port()
```

Comment:
services already have this logic, because it's inherited [from LocalProcessSpawner here](https://github.com/jupyterhub/jupyterhub/blob/e255ada1691a8300509085238707b0e21af7c4d5/jupyterhub/spawner.py#L1659-L1660).

Reply:
You override the `start()` function in this class, and don't call `super.start()` anywhere.

Hence this is why the tests started failing, and I copied the logic into this class: https://github.com/jupyterhub/jupyterhub/actions/runs/4005957446/jobs/6876931281#step:10:421

Reply:
Ah, forgot about the lack of super start. Then this change is definitely correct. Thanks!